src/main/java/com/simpligility/maven/plugins/android/phase01generatesources/GoogleServicesProcessor.java

```diff
@@ -0,0 +1,488 @@
+package com.simpligility.maven.plugins.android.phase01generatesources;
+
+import com.google.common.base.Strings;
+import com.google.common.io.Files;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonPrimitive;
+import org.apache.maven.plugin.logging.Log;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Map;
+import java.util.TreeMap;
+
+import static com.google.common.base.Charsets.UTF_8;
+
+/**
+ * This code is based on the corresponding Gradle plugin. I have tried to keep the differences to a minimum (apart from
```

Comment:
But what does the class do?
A class comment should describe the responsibility of the class.